### PR TITLE
Added Freetype font support

### PR DESCRIPTION
**Pull request by PrinceDeveloperOf**
Added support for freetype fonts. The fonts are rendered and put into a texture at initialization. They're derived from BMFonts, and work in the same way. The name for a particular font, is the filename plus pt size. It uses the filename plus the pt size to orginize different sized fonts. 

And now stb_truetype is used intead of freetype.

### CODE REVIEW

**Comment by elnormous** [comment]
Please fix all the errors and warnings: https://ci.appveyor.com/project/elnormous/ouzel/build/1.0.2168

**Comment by PrinceDeveloperOf** [comment]
Errors fixed.

**Comment by elnormous** [comment]
Could you please merge the master into this branch and solve the conflicts?

File: build/jni/Android.mk [hunk 1]
```diff
@@ -72,6 +72,7 @@ LOCAL_SRC_FILES :=../../ouzel/animators/Animator.cpp \
     ../../ouzel/graphics/TextureResource.cpp \
     ../../ouzel/graphics/Vertex.cpp \
     ../../ouzel/gui/BMFont.cpp \
+	../../ouzel/gui/FMFont.cpp \
```
Comment (elnormous) [inline build/jni/Android.mk:77]: This should be FTFont.cpp?

**Comment by elnormous** [comment]
Please convert all tabs to 4 spaces.

**Comment by elnormous** [comment]
Use std::copy instead of memcpy

**Comment by elnormous** [comment]
Use unsigned integer (uint16_t) for font size, because the size can not be negative.

File: build/jni/Android.mk (hunk 1 above)
Comment (PrinceDeveloperOf) [reply build/jni/Android.mk:77]: Yeah that should be.

**Comment by PrinceDeveloperOf** [comment]
All uses of memcpy have turned to std::copy, font size is now represented as a uint16_t, and all tabs should be converted.

**Comment by elnormous** [comment]
Please merge the master into this pull request.

**Pull request merged**